apps/dh/api-dh/source/DataHub.WebApi/Modules/ImbalancePrice/ImbalancePriceNode.cs

```diff
@@ -98,78 +101,13 @@ public async Task<IEnumerable<ImbalancePricesDailyDto>> GetImbalancePricesForMon
                 continue;
             }
 
-            var missingTimestamps = FindMissingTimestaps(imbalancePrice);
-
             imbalancePrice.ImbalancePrices = imbalancePrice.ImbalancePrices
-                .Concat(missingTimestamps)
+                .Concat(imbalancePrice.FindMissingTimestamps())
                 .OrderBy(x => x.Timestamp)
-                .ToList();
-
-            EnsureFullDataset(imbalancePrice.ImbalancePrices);
+                .ToList()
+                .EnsureFullDataset();
```

Comment:
suggestion: create a local `ImbalancePriceClient` instead of hiding this functionality in extensions methods

Reply:
Moved logic into client